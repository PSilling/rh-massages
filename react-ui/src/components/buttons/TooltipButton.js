// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * An outline button component with an attached tooltip.
 */
class TooltipButton extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  render() {
    const { disabled, label, tooltip, ...rest } = this.props;
    return (
      <span className="no-print">
        <Button {...rest} id={this.tooltipTarget} outline disabled={this.props.disabled}>
          {this.props.label}
        </Button>
        {this.props.tooltip !== "" && !this.props.disabled && (
          <Tooltip
            isOpen={this.state.tooltipActive}
            target={this.tooltipTarget}
            toggle={this.toggleTooltip}
            trigger="hover"
          >
            {this.props.tooltip}
          </Tooltip>
        )}
      </span>
    );
  }
}

TooltipButton.propTypes = {
  /** function to be called on button click */
  onClick: PropTypes.func.isRequired,
  /** whether the button should be displayed as active or not */
  active: PropTypes.bool,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button label to be displayed */
  label: PropTypes.string,
  /** tooltip shown over the button */
  tooltip: PropTypes.string
};

TooltipButton.defaultProps = {
  active: false,
  disabled: false,
  label: "Click me",
  tooltip: ""
};

export default TooltipButton;
