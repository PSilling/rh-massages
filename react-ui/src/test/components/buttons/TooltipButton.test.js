// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import { Button, Tooltip } from "reactstrap";
import TooltipButton from "../../../components/buttons/TooltipButton";

test("renders content with correct props", () => {
  const testFunction = jest.fn();
  const wrapper = shallow(<TooltipButton onClick={testFunction} label="test" tooltip="tooltip" />);
  const button = wrapper.find(Button);
  const tooltip = wrapper.find(Tooltip);

  button.props().onClick();

  expect(testFunction).toHaveBeenCalledTimes(1);
  expect(button.props().onClick).toBe(testFunction);
  expect(button.props().children).toEqual("test");
  expect(button.props().disabled).toBe(false);
  expect(tooltip.props().isOpen).toBe(false);
  expect(tooltip.props().children).toEqual("tooltip");
  expect(tooltip.props().target).toEqual(button.props().id);
  expect(wrapper).toMatchSnapshot();
});
