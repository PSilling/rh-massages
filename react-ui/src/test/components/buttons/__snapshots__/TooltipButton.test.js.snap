// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TooltipButton
    active={false}
    disabled={false}
    label="test"
    onClick={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    tooltip="tooltip"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Button
          active={false}
          color="secondary"
          disabled={false}
          id="Tooltip1"
          onClick={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
          outline={true}
          tag="button"
        >
          test
        </Button>,
        <Tooltip
          autohide={true}
          delay={
            Object {
              "hide": 250,
              "show": 0,
            }
          }
          hideArrow={false}
          isOpen={false}
          placement="top"
          placementPrefix="bs-tooltip"
          target="Tooltip1"
          toggle={[Function]}
          trigger="hover"
        >
          tooltip
        </Tooltip>,
      ],
      "className": "no-print",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "active": false,
          "children": "test",
          "color": "secondary",
          "disabled": false,
          "id": "Tooltip1",
          "onClick": [MockFunction] {
            "calls": Array [
              Array [],
            ],
            "results": Array [
              Object {
                "isThrow": false,
                "value": undefined,
              },
            ],
          },
          "outline": true,
          "tag": "button",
        },
        "ref": null,
        "rendered": "test",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autohide": true,
          "children": "tooltip",
          "delay": Object {
            "hide": 250,
            "show": 0,
          },
          "hideArrow": false,
          "isOpen": false,
          "placement": "top",
          "placementPrefix": "bs-tooltip",
          "target": "Tooltip1",
          "toggle": [Function],
          "trigger": "hover",
        },
        "ref": null,
        "rendered": "tooltip",
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            active={false}
            color="secondary"
            disabled={false}
            id="Tooltip1"
            onClick={
              [MockFunction] {
                "calls": Array [
                  Array [],
                ],
                "results": Array [
                  Object {
                    "isThrow": false,
                    "value": undefined,
                  },
                ],
              }
            }
            outline={true}
            tag="button"
          >
            test
          </Button>,
          <Tooltip
            autohide={true}
            delay={
              Object {
                "hide": 250,
                "show": 0,
              }
            }
            hideArrow={false}
            isOpen={false}
            placement="top"
            placementPrefix="bs-tooltip"
            target="Tooltip1"
            toggle={[Function]}
            trigger="hover"
          >
            tooltip
          </Tooltip>,
        ],
        "className": "no-print",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "active": false,
            "children": "test",
            "color": "secondary",
            "disabled": false,
            "id": "Tooltip1",
            "onClick": [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            },
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autohide": true,
            "children": "tooltip",
            "delay": Object {
              "hide": 250,
              "show": 0,
            },
            "hideArrow": false,
            "isOpen": false,
            "placement": "top",
            "placementPrefix": "bs-tooltip",
            "target": "Tooltip1",
            "toggle": [Function],
            "trigger": "hover",
          },
          "ref": null,
          "rendered": "tooltip",
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
